src/toxcore/tcp/server/server.rs

```diff
@@ -603,10 +673,50 @@ mod tests {
         ));
     }
     #[test]
+    fn handle_onion_request() {
+        let (tcp_onion_sink, tcp_onion_stream) = mpsc::unbounded();
+        let server = Server::new(tcp_onion_sink);
+        let (client_pk, _, _) = add_random_client(&server);
+
+        let request = OnionRequest {
+            nonce: gen_nonce(),
+            ip_port: IpPort {
+                addr: IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
+                port: 12345
+            },
+            data: [13; 100].to_vec()
+        };
+        let handle_res = server
+            .handle_packet(&client_pk, Packet::OnionRequest(request.clone()))
+            .wait();
+        assert!(handle_res.is_ok());
+
+        let (packet, _) = tcp_onion_stream.into_future().wait().unwrap();
+        assert_eq!(packet.unwrap(), request);
+    }
```

Comment:
Add `handle_onion_response` and check error. See https://coveralls.io/builds/15364599/source?filename=src%2Ftoxcore%2Ftcp%2Fserver%2Fserver.rs#L89 for missing lines.